File: 1/2021.03.27/230.Kth Smallest Element in a BST/python/solution.py
class Solution:
    def kthSmallest(self, root: TreeNode, k: int) -> int:
        def inorder(root):
            if root:
                inorder(root.left)
                if self.k == 1:
                    self.res = root.val
                self.k -= 1
                inorder(root.right)
        
        self.k = k
        inorder(root)
        return self.res    

File: 1/2021.03.27/230.Kth Smallest Element in a BST/java/solution.java
class Solution {
    
    int cur = 0;
    int res = 0;
    public int kthSmallest(TreeNode root, int k) {
        inorder(root, k);
        return res;
    }
    
    private void inorder(TreeNode root, int k) {
        if (root != null) {
            inorder(root.left, k);
            cur++;
            if (cur == k) {
                res = root.val;
                return;
            }
            inorder(root.right, k);
        }
    } 
}
